source/Buoy-Development-Tools/TonelWriterV3.extension.st

```diff
@@ -0,0 +1,26 @@
+Extension { #name : 'TonelWriterV3' }
+
+{ #category : '*Buoy-Development-Tools' }
+TonelWriterV3 >> typeClassDefinitionOf: aClassDefinition [
+
+	| definition |
+	definition := OrderedDictionary new.
+	self
+		at: #name put: aClassDefinition className in: definition;
+		at: #superclass put: aClassDefinition superclassName in: definition.
+	aClassDefinition type = #normal ifFalse: [ self at: #type put: aClassDefinition type in: definition ].
+	aClassDefinition hasTraitComposition ifTrue: [ definition at: #traits put: aClassDefinition traitCompositionString ].
+	aClassDefinition hasClassTraitComposition ifTrue: [ definition at: #classTraits put: aClassDefinition classTraitCompositionString ].
+	aClassDefinition instVarNames ifNotEmpty: [ :vars | definition at: #instVars put: vars asArray ].
+	(aClassDefinition variables
+		 select: #isClassVariable
+		 thenCollect: #name) ifNotEmpty: [ :vars | definition at: #classVars put: vars asArray ].
+	(aClassDefinition variables
+		 select: #isPoolImport
+		 thenCollect: #name) ifNotEmpty: [ :vars | definition at: #pools put: vars asArray ].
+	aClassDefinition classInstVarNames ifNotEmpty: [ :vars | definition at: #classInstVars put: vars asArray ].
+	self setPackageInfoOf: aClassDefinition in: definition.
+	"Write gs_options if available as class properties"
+	(aClassDefinition actualClass propertyAt: #gs_options) ifNotNil: [ :options | definition at: #gs_options put: options ].
+	^ self toSTON: definition
```

Comment:
This change seems to overwrite an existing method. We could add some test to check that the original code is the one expected here. Also, the code looks like it's missing BA-ST formatting.

Reply:
I will not format this method with our conventions on purpose, because it's a modified version of the superclass one with some additions and it's easier to compare if we don't change the formatting.
Sadly, the `TonelWriter` hierarchy does not provide any nice hook to extend it with additional properties for the class.